packages/mui-material-next/src/InputBase/InputBase.tsx

```diff
@@ -0,0 +1,727 @@
+    //   muiFormControl.onBlur(event);
+    // }
+  };
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    ...args: any[]
+  ) => {
+    if (!isControlled && event.target != null) {
+      checkDirty({
+        value: event.target.value,
+      });
+    }
+
+    // Perform in the willUpdate
+    if (onChange) {
+      // @ts-ignore
+      onChange(event, ...args);
+    }
+  };
+
+  const handleClick = (event: React.PointerEvent) => {
+    if (onClick /* && !fcs.disabled */) {
+      onClick(event);
+    }
+  };
+
+  React.useEffect(() => {
+    if (muiFormControl) {
+      muiFormControl.setAdornedStart(Boolean(startAdornment));
+    }
+  }, [muiFormControl, startAdornment]);
+
+  const {
+    getRootProps,
+    getInputProps,
+    focused: focusedState,
+    // TODO: what if this `formControlContext` is completely ignored and the component uses a completely separate one similar to Joy
+    // formControlContext,
+    error: errorState,
+    disabled: disabledState,
+    inputRef,
+  } = useInput({
+    disabled: disabledProp,
+    defaultValue,
+    error,
+    onBlur: handleBlur,
+    onClick: handleClick,
+    onChange: handleChange,
+    onFocus: handleFocus,
+    required,
+    value,
+    inputRef: inputRefProp,
+  });
+
+  // TODO: integrate ownerState properties with material-next/FormControl: color, disabled, error, focused, hiddenLabel, size
+  const ownerState = {
+    ...props,
+    color: colorProp || 'primary',
+    disabled: disabledState,
+    endAdornment,
+    error: errorState,
+    focused: focusedState,
+    formControl: muiFormControl,
+    fullWidth,
+    hiddenLabel: false, // TODO: material-next/FormControl integration
+    multiline,
+    size: sizeProp,
+    startAdornment,
+    type,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  const propsToForwardToInputSlot = {
+    'aria-describedby': ariaDescribedby,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby,
+    autoComplete,
+    autoFocus,
+    id,
+    onKeyDown,
+    onKeyUp,
+    name,
+    placeholder,
+    readOnly,
+    type,
+  };
+
+  const Root = slots.root || InputBaseRoot;
+  const rootProps: WithOptionalOwnerState<InputBaseRootSlotProps> = useSlotProps({
+    elementType: Root,
+    getSlotProps: getRootProps,
+    externalSlotProps: slotProps.root,
+    externalForwardedProps: other,
+    additionalProps: {
+      ref: forwardedRef,
+    },
+    ownerState,
+    className: [classes.root, className],
+  });
+
+  const InputComponent = multiline
+    ? slots.textarea ?? TextareaAutosize
+    : slots.input ?? InputBaseInput;
+
+  const {
+    ownerState: inputOwnerState,
+    ...otherInputProps
+  }: WithOptionalOwnerState<InputBaseInputSlotProps> = useSlotProps({
+    elementType: InputComponent,
+    getSlotProps: (otherHandlers: EventHandlers) => {
+      return getInputProps({
+        ...propsToForwardToInputSlot,
```

Comment:
I think there might be an inconsistency between `useButton` and `useInput` that might be worth revisiting. The `getRootProps` function of `useButton` is [typed to receive handlers](https://github.com/mj12albert/material-ui/blob/b26f4d698c7492c2f432689da0a79b92f524591b/packages/mui-base/src/useButton/useButton.ts#L208), but the `getRootProps` and `getInputProps` functions of `useInput` [receive any prop](https://github.com/mj12albert/material-ui/blob/b26f4d698c7492c2f432689da0a79b92f524591b/packages/mui-base/src/useInput/useInput.ts#L167). I think that might be confusing for users that use both. What do you think @michaldudak?

Reply:
@DiegoAndai I think accepting any prop is the way to go ~ there's an issue about this here: https://github.com/mui/material-ui/issues/38186